script.js

```diff
@@ -1,4 +1,123 @@
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+//creating helper function to roll 2 dice
+var rollDiceForPlayer = function () {
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+  //after this step, ask player to choose the order
+  console.log(currentPlayerRolls[0]);
+  console.log(currentPlayerRolls[1]);
+  return `Welcome Player ${currentPlayer}! <br><br> You rolled ${currentPlayerRolls[0]} for Dice 1 and ${currentPlayerRolls[1]} for Dice 2. <br><br> Please choose the dice number (either 1 or 2) to be your first digit of your final value.`;
+};
+
+var getPlayerScore = function (playerInput) {
+  // input validation
+  if (playerInput != 1 && playerInput != 2) {
+    return `Error! Please only input "1" or "2" to choose your first digit of your final value. <br> Dice 1: ${currentPlayerRolls[0]} | Dice 2: ${currentPlayerRolls[1]}`;
+  }
+
+  // input == 1
+  if (playerInput == 1) {
+    var playerFinalValue = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+
+  // input == 2
+  if (playerInput == 2) {
+    var playerFinalValue = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
+  // store current player score
+  allPlayerSCore.push(playerFinalValue);
+  console.log(playerFinalValue);
+  // clear current player rolls array
+  currentPlayerRolls = [];
+  return `Player ${currentPlayer}, your final value is ${playerFinalValue}.`;
+};
+
+var comparePlayersScores = function () {
+  // have to use another outputvalue
+  var compareOutputValue = `Player 1 score: ${allPlayerSCore[0]} <br> Player 2 score: ${allPlayerSCore[1]}`;
+
+  // player 1 wins
+  if (allPlayerSCore[0] > allPlayerSCore[1]) {
+    compareOutputValue = `${compareOutputValue} <br><br> Player 1 wins!`;
+  }
+  //player 2 wins
+  if (allPlayerSCore[0] < allPlayerSCore[1]) {
+    compareOutputValue = `${compareOutputValue} <br><br> Player 2 wins!`;
+  }
+  //tie
+  if (allPlayerSCore[0] == allPlayerSCore[1]) {
+    compareOutputValue = `${compareOutputValue} <br><br> It's a tie!`;
+  }
+  return compareOutputValue;
+};
+
+//reset game helper function
+var resetGame = function () {
+  currentPlayer = 1;
+  currentGameMode = gameModeDiceRoll;
+  allPlayerSCore = [];
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log(currentPlayer);
+  var myOutputValue = "";
+
+  if (currentGameMode == gameModeDiceRoll) {
+    // set the name
+    // userName = input;
+    // now that we have the name for player 1 and rolled 2 dices, switch the mode
+    currentGameMode = gameStateChooseDiceOrder;
+    console.log(currentGameMode);
+    return (myOutputValue = rollDiceForPlayer());
+  }
+
+  if (currentGameMode == gameStateChooseDiceOrder) {
+    // call getPlayerScore helper function
+    myOutputValue = getPlayerScore(input);
```

Comment:
theres a bug in this line

![image](https://user-images.githubusercontent.com/40411953/156780816-0122ce07-131a-42de-9ff2-8e44e3da74c6.png)

your input validation for `getPlayerScore` will return out of `getPlayerScore` with your input validation msg but after it is returned to `main`, the `main` function doesn't stop running because you didn't specify it to

thats why in the above screenshot, the gameplay still moves on to player 2

Reply:
then when the gameplay moves on to player 2 and i play through it properly, player 2's score will be stored as player 1's score and i get this eventual result 

![image](https://user-images.githubusercontent.com/40411953/156781268-4a674d7b-cabb-49c7-9480-2afaf0444137.png)

Reply:
consider how you can change this up so that the main function stops running here if the input is not valid (hint: maybe shift the input validation out of `getPlayerScore` and into `main`, then return out of `main` with the relevant msg if the input is not valid)